tests/test_merge.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-FileCopyrightText: 2023 TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from unittest import TestCase, mock
+
+import pytest
+
+from opossum_lib.merger import (
+    _merge_dicts_without_duplicates,
+    _merge_resources,
+    _merge_resources_to_attributions,
+    expand_opossum_package_identifier,
+    merge_opossum_information,
+)
+from opossum_lib.opossum_file import (
+    Metadata,
+    OpossumInformation,
+    OpossumPackage,
+    Resource,
+    SourceInfo,
+)
+
+
+@mock.patch("opossum_lib.opossum_file.OpossumPackage", autospec=True)
+def test_merge_opossum_information(opossum_package: OpossumPackage) -> None:
+    opossum_information = OpossumInformation(
+        Metadata("project-id", "30-05-2023", "test data"),
+        Resource({"A": Resource({"B": Resource({})})}),
+        {"SPDXRef-Package": opossum_package},
+        {"/A/B/": ["SPDXRef-Package"]},
+    )
+
+    opossum_information_2 = OpossumInformation(
+        Metadata("test-data-id", "29-05-2023", "second test data"),
+        Resource({"A": Resource({"D": Resource({"C": Resource({})})})}),
+        {"SPDXRef-File": opossum_package},
+        {"/A/D/C/": ["SPDXRef-File"]},
+    )
+
+    merged_information = merge_opossum_information(
+        [opossum_information, opossum_information_2]
+    )
+
+    assert merged_information.metadata == opossum_information.metadata
+    assert merged_information.resources == Resource(
+        {"A": Resource({"B": Resource({}), "D": Resource({"C": Resource({})})})}
+    )
+    assert merged_information.externalAttributions == {
+        "project-id-SPDXRef-Package": opossum_package,
+        "test-data-id-SPDXRef-File": opossum_package,
+    }
+    assert merged_information.resourcesToAttributions == {
+        "/A/B/": ["project-id-SPDXRef-Package"],
+        "/A/D/C/": ["test-data-id-SPDXRef-File"],
+    }
+
+
+def test_merge_resources() -> None:
+    list_of_paths = [["A"], ["A", "B", "C"], ["A", "B"], ["A", "D"]]
+    resource = Resource()
+
+    for path in list_of_paths:
+        resource.add_path(path)
+    list_of_paths = [["C", "D", "E"], ["A", "B", "C"], ["A", "B"], ["A", "D"]]
+    resource2 = Resource()
+
+    for path in list_of_paths:
+        resource2.add_path(path)
+    resources = [resource, resource2]
+    merged_resource = _merge_resources(resources)
+
+    assert merged_resource == Resource(
+        {
+            "A": Resource({"B": Resource({"C": Resource({})}), "D": Resource({})}),
+            "C": Resource({"D": Resource({"E": Resource({})})}),
+        }
+    )
+
+
+def test_merge_resources_to_attributions() -> None:
+    resources_to_attributions = [
+        {"resources/Path": ["identifier-A", "identifier-B"]},
+        {"resources/Path": ["identifier-C"]},
+        {"resources/Path/different": ["identifier-A"]},
+    ]
+    resources_to_attribution = _merge_resources_to_attributions(
```

Comment:
Naming could be a bit confusing here as `resources_to_attribution` indicates that a resource can only be assigned with a single attribution. Maybe something like `merged_resources_to_attributions` is clearer.

Reply:
Good catch, I changed the name accordingly.